rfc.md

```diff
@@ -0,0 +1,215 @@
+   It's useful to be able to read the phenotype of an organism easily, e.g.
+   `createOrganism(...).phenotype.color` instead of
+   `org = createOrganism(...); getCharacteristic(org, 'wings')`, so it seems to make
+   sense for this to be a part of any organism generated by the library. However, it
+   is redundant information, so it probably doesn't need to be passed in when using
+   an organism.
+* secondXAlleles: string  
+   This formalizes the pattern from geni* that allows us to switch sexes without changing the
+   phenotype. **Maybe this is unneeded?** We could presumably also have a `changeSex` function
+   that explictly picks second-x-alleles that don't change the phenotype. Do we need this for
+   anything else?
+
+### Creation
+```
+  organism({
+    /** optional. One would pass in fully-specified alleles when re-generating
+      phenotypes */
+    alleles: "a-a b-b ..."
+    /** optional. Explicitly specifying the authored syntax */
+    authoredAlleles: "a-a b- -c d-[D1, D2] e-^[E1] XY"
+    /** optional */
+    sex: "male"
+  }): Organism
+
+  breed(org: Organism, org: Organism, crossover=true): Organism
+
+  breed(org: Organism, org: Organism, quantity: number, crossover=true): Organism[]
+
+  fertilize(gamete: Gamete, gamete: Gamete): Organism
+```
+
+## Problem: Species
+
+In biologica 1.0, we need to pass in a BioLogica.Species object in any Organism creation
+method. This seems ugly, and would be even more so now because we'd no longer create
+Organism classes that have their own references to the species, so we'd have to *either*
+add the entire species spec to every organism, *or* pass in the species spec in every
+single method (e.g. `breed(Organism, Organism, Species)`).
+
+One solution would be to have an initialization method that sets up the species from the start:
+
+```
+  import biologica from `biologica`
+
+  const Drakes = biologica(drakeSpecies)
+  const org1 = Drakes.organism(...)
+  const child = Drakes.breed(org1, org2)
+
+  // instead of
+
+  import { organism, breed } from `biologica`
+
+  const org1 = organism(..., drakeSpecies)
+  const child = breed(org1, org2, drakeSpecies)
+```
+The top example seems a little uglier than the alternative below, but saves us from
+including `drakeSpecies` everywhere. Thoughts?
+
+Edit: Or maybe we can keep the top solution but make it cleaner by using
+
+```
+  import biologica from `biologica`
+  const { organism, breed } = biologica(drakeSpecies)
+
+  const org1 = organism(...)
+  const child = breed(org1, org2)
+```
+
+## Small problem of not using classes
+
+This is probably not so important, but classes with methods can often be read in a
+more literate manner: If we wanted to, say, get the image src for an authored organism
+in 1.0, we could write
+
+`new Organism(authoredAlleles).getImageSrc()`
+
+In 2.0 it seems that we would either need to split it into two lines or use the backwards
+
+`getImageSrc(organism(authoredAlleles))`
```

Comment:
I think we could probably preserve the advantages of plain objects while also providing some of the syntactic conveniences of .methods(). MobX State Tree does something similar, for one example.

Reply:
One additional advantage of having the computed properties (phenotype, imageSrc) be simple properties of the object, instead of methods, is that if we *do* want to serialize the organism with all the properties, it provides a complete standard way to do so.

Currently, for instance, the ITS would like the phenotype of organisms submitted by the user. This is put in a custom property set up for the ITS because the Organism doesn't serialize nicely, so we have to set each property separately on the sent object. This is fine, but if we do it differently in one place than another, it could lead to confusion.